src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.CompilationTracker.cs

```diff
@@ -490,12 +490,16 @@ private static string LogBuildCompilationAsync(ProjectState state)
                 {
                     var compilation = CreateEmptyCompilation();
 
+                    var trees = new SyntaxTree[ProjectState.DocumentIds.Count];
```

Comment:
worth avoiding the allocation by using an ArrayBuilder+Free?

Reply:
➡️ The target of the invocation has a `params SyntaxTree[]` parameter, so without making this change impact signatures this approach is fairly ideal. The profiling report indicated there is little left to gain with respect to this specific change.

Reply:
> The target of the invocation has a params SyntaxTree[] parameter, so without making this change impact signatures this approach is fairly ideal

Yes.  but it all calls through to the IEnumerable overload.  It seems like it would just be idiomatic and cleaner to just use ArrayBuilder+Add like we do elsewhere.

Reply:
NBD though.  If you prefer this, by all means keep it :)